### PR TITLE
adding timeout and more logs to get python version

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

**Review by tkatzir — APPROVED** [review]
Great work!

**Review by yaakovi — APPROVED** [review]
Logs should be in debug, not info

File: demisto_sdk/commands/lint/docker_helper.py [hunk 1]
```diff
@@ -36,6 +36,13 @@ def init_global_docker_client(timeout: int = 60, log_prompt: str = ''):
         except Exception:
             logger.exception(f'{log_prompt} - failed to login to docker registry')
 
+    else:
+        msg = 'docker client already available, using current DOCKER_CLIENT'
+        if log_prompt:
+            logger.info(f'{log_prompt} - {msg}')
+        else:
+            logger.info(msg)
```
Comment (dorschw) [inline demisto_sdk/commands/lint/docker_helper.py:44]:
```suggestion
            logger.info(f'{log_prompt} - {msg}' if log_prompt else msg)
```

Comment (dorschw) [inline demisto_sdk/commands/lint/docker_helper.py:44]:
```suggestion
    else:
        msg = 'docker client already available, using current DOCKER_CLIENT'
        logger.info(f'{log_prompt} - {msg}' if log_prompt else msg)
```

File: demisto_sdk/commands/lint/helpers.py [hunk 2]
```diff
@@ -303,28 +303,34 @@ def get_python_version_from_image(image: str, timeout: int = 60) -> str:
 
     py_num = '3.8'
     # Run three times
-    log_prompt = 'Get python version from image'
+    log_prompt = f'Get python version from image {image} with {timeout}'
```
Comment (esharf) [inline demisto_sdk/commands/lint/helpers.py:306]:
```suggestion
    log_prompt = f'Get python version from image {image}'
```

**Review by esharf — APPROVED** [review]

File: demisto_sdk/commands/lint/helpers.py [hunk 3]
```diff
@@ -300,43 +300,31 @@ def get_python_version_from_image(image: str, timeout: int = 60) -> str:
     match_group = re.match(r'[\d\w]+/python3?:(?P<python_version>[23]\.\d+)', image)
     if match_group:
         return match_group.groupdict()['python_version']
-
-    py_num = '3.8'
+    py_num = None
     # Run three times
-    log_prompt = 'Get python version from image'
+    log_prompt = f'Get python version from image {image}'
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:305]:
```suggestion
    log_prompt = f'Getting python version from {image=}'
```

File: demisto_sdk/commands/lint/helpers.py [hunk 4]
```diff
@@ -300,43 +300,31 @@ def get_python_version_from_image(image: str, timeout: int = 60) -> str:
...
+        # Wait for container to finish
+        logger.debug(f'{log_prompt} - Container finished running.')
+
+        # Get python version
+        py_num = parse(py_num.decode("utf-8")).base_version
+        
+    except Exception:
+        logger.exception(f'{log_prompt} - Failed detecting Python version for image {image}')
+    logger.info(f'{log_prompt} - End. Python version is {py_num}')
+    return py_num if py_num else '3.10'
```
Comment (dorschw) [inline demisto_sdk/commands/lint/helpers.py:327]: is this default safe to assume?

**Pull request merged**